ts/sagas/onboarding.ts

```diff
@@ -97,11 +105,42 @@ function* tosCheckSaga(): Iterator<Effect> {
     });
     yield put(navigateToOnboardingTosScreenAction);
 
-    // Here we wait the user accept the ToS
-    yield take(TOS_ACCEPT_REQUEST);
+    // Loop until this step is fulfilled: TOS accepted and if necessary Profile upserted.
+    while (true) {
+      // Here we wait the user accept the ToS
+      yield take(TOS_ACCEPT_REQUEST);
+
+      // Get the current user profile from the store
+      const userProfile: ProfileState = yield select(profileSelector);
+
+      // We have the profile info but the user hasn't a profile active on API
```

Comment:
`hasn't a profile active on API` -> `doesn't yet have an active profile in the CD APIs`